monitor/internal/pod/delete_controller.go

```diff
@@ -127,6 +162,52 @@ func deleteControllerProcessItem(backgroundCtx context.Context, c client.Client,
 		}
 		// we only fire events away, we don't really care about the error anyway
 		// it is up to the policy engine to make sense of that
-		delete(*m, nativeID)
+		delete(m, nativeID)
+		return
+	}
+
+	// now the 2nd case, when pod UID match
+	if string(pod.UID) == delObj.nativeID {
+		zap.L().Info("Delete-controller: the pod UID Match happened, delete the", zap.String("podName:", req.String()), zap.String("podUID", string(pod.UID)))
+		// 2a get the current sandboxID
+		currentSandboxID, err := sandboxExtractor(ctx, pod)
+		if err != nil {
+			zap.L().Error("Delete controller, cannot extract the SandboxID, return", zap.String("puID", nativeID), zap.String("namespacedName", req.String()), zap.String("podUID", string(pod.GetUID())))
+			return
+		}
+		// update the map with the sandboxID
+		if delObj.sandboxID == "" {
+			updatedDelObj := DeleteObject{nativeID: nativeID, sandboxID: currentSandboxID, podName: req}
+			m[nativeID] = updatedDelObj
+		}
+		// 2b get the pod/old sandboxID
+		if delObj, ok = m[nativeID]; !ok {
```

Comment:
you already got the delObj at the beginning of the function, no need for a 2nd retrieval and error check